Makefile

```diff
@@ -0,0 +1,105 @@
+RUST_TOOLCHAIN_NIGHTLY = nightly-2025-02-16
+SOLANA_CLI_VERSION = 2.3.4
+
+nightly = +${RUST_TOOLCHAIN_NIGHTLY}
+
+make-path = $1
+
+rust-toolchain-nightly:
+	@echo ${RUST_TOOLCHAIN_NIGHTLY}
+
+solana-cli-version:
+	@echo ${SOLANA_CLI_VERSION}
+
+cargo-nightly:
+	cargo $(nightly) $(ARGS)
+
+audit:
+	cargo audit \
+			--ignore RUSTSEC-2022-0093 \
+			--ignore RUSTSEC-2024-0421 \
+			--ignore RUSTSEC-2024-0344 \
+			--ignore RUSTSEC-2024-0376 $(ARGS)
+
+spellcheck:
+	cargo spellcheck --code 1 $(ARGS)
+
+clippy-%:
+	cargo $(nightly) clippy --manifest-path $(call make-path,$*)/Cargo.toml \
+	  --all-targets \
+	  --all-features \
+		-- \
+		--deny=warnings \
+		--deny=clippy::default_trait_access \
+		--deny=clippy::arithmetic_side_effects \
+		--deny=clippy::manual_let_else \
+		--deny=clippy::used_underscore_binding $(ARGS)
+
+format-check-%:
+	cargo $(nightly) fmt --check --manifest-path $(call make-path,$*)/Cargo.toml $(ARGS)
+
+powerset-%:
+	cargo $(nightly) hack check --feature-powerset --all-targets --manifest-path $(call make-path,$*)/Cargo.toml $(ARGS)
+
+semver-check-%:
+	cargo semver-checks --manifest-path $(call make-path,$*)/Cargo.toml $(ARGS)
+
+shellcheck:
+	git ls-files -- '*.sh' | xargs shellcheck --color=always --external-sources --shell=bash $(ARGS)
+
+sort-check:
+	cargo $(nightly) sort --workspace --check $(ARGS)
+
+bench-%:
+	cargo $(nightly) bench --manifest-path $(call make-path,$*)/Cargo.toml $(ARGS)
+
+format-rust:
+	cargo $(nightly) fmt --all $(ARGS)
+
+build-sbf-%:
+	cargo build-sbf --manifest-path $(call make-path,$*)/Cargo.toml $(ARGS)
+
+build-wasm-%:
+	cargo build --target wasm32-unknown-unknown --manifest-path $(call make-path,$*)/Cargo.toml --all-features $(ARGS)
+
+build-doc-%:
+	RUSTDOCFLAGS="--cfg docsrs -D warnings" cargo $(nightly) doc --all-features --no-deps --manifest-path $(call make-path,$*)/Cargo.toml $(ARGS)
+
+test-doc-%:
+	cargo $(nightly) test --doc --all-features --manifest-path $(call make-path,$*)/Cargo.toml $(ARGS)
+
+test-%:
+	SBF_OUT_DIR=$(PWD)/target/deploy cargo $(nightly) test --manifest-path $(call make-path,$*)/Cargo.toml $(ARGS)
+
+format-check-js-%:
+	cd $(call make-path,$*) && pnpm install && pnpm format $(ARGS)
+
+lint-js-%:
+	cd $(call make-path,$*) && pnpm install && pnpm lint $(ARGS)
+
+test-js-%:
```

Comment:
Ordering problem:  the earlier patterns `format-check-` and `test-` match first, so they look for non-existent directories. 

i.e.:

```
❯ make test-js-type-length-value-js
error: manifest path `js-type-length-value-js/Cargo.toml` does not exist
make: *** [test-js-type-length-value-js] Error 101
```

Solution: 
- move `format-check-js-%` and `test-js-%` before `format-check-%` and `test-%`